Mods/SML/Source/SML/Private/Patching/BlueprintHookManager.cpp

```diff
@@ -12,18 +12,18 @@ DEFINE_LOG_CATEGORY(LogBlueprintHookManager);
 
 #define WRITE_UNALIGNED(Arr, Type, Value) \
 	Arr.AddUninitialized(sizeof(Type)); \
-	FPlatformMemory::WriteUnaligned<Type>(&AppendedCode[Arr.Num() - sizeof(Type)], (Type) Value);
+	FPlatformMemory::WriteUnaligned<Type>(&Arr[Arr.Num() - sizeof(Type)], (Type) Value);
 
 void UBlueprintHookManager::HandleHookedFunctionCall(FFrame& Stack, int32 HookOffset) {
 	FFunctionHookInfo& FunctionHookInfo = HookedFunctions.FindChecked(Stack.Node);
 	FunctionHookInfo.InvokeBlueprintHook(Stack, HookOffset);
 }
 
 #if DEBUG_BLUEPRINT_HOOKING
-void DebugDumpFunctionScriptCode(UFunction* Function, int32 HookOffset, const FString& Postfix) {
-	const FString FileLocation = FPaths::RootDir() + FString::Printf(TEXT("BlueprintHookingDebug_%s_%s_at_%d_%s.json"), *Function->GetOuter()->GetName(), *Function->GetName(), HookOffset, *Postfix);
+void DebugDumpFunctionScriptCode(UFunction* Function, int32 HookOffset, int32 ResolvedHookOffset, const FString& Postfix) {
+	const FString FileLocation = FPaths::RootDir() + FString::Printf(TEXT("BlueprintHookingDebug_%s_%s_at_%d_resolved_%d_%s.json"), *Function->GetOuter()->GetName(), *Function->GetName(), HookOffset, ResolvedHookOffset, *Postfix);
 
-	FKismetBytecodeDisassemblerJson Disassembler;
+	FSMLKismetBytecodeDisassembler Disassembler;
```

Comment:
Any reason the type of this got changed? Did it not build before?

Reply:
I asked Arch about this when I first started debugging the crash - the type name was changed at some point and this code never got updated because it was only compiled in when DEBUG_BLUEPRINT_HOOKING is set to 1 and apparently nobody had tried to test it in quite a while.

Reply:
Ack